src/qibolab/result.py

```diff
@@ -5,97 +5,200 @@
+        return np.angle(self.voltage.i + 1.0j * self.voltage.q)
+
+    # We are asumming results from the same experiment so same number of shots
+    def __add__(self, data):  # __add__(self, data:IQResults) -> IQResults
+        axis = 0
+        i = np.append(self.voltage.i, data.voltage.i, axis=axis)
+        q = np.append(self.voltage.q, data.voltage.q, axis=axis)
+        return IQResults(i, q)
+
+    def serialize(self):
+        """Serialize as a dictionary."""
+        serialized_dict = {
+            "magnitude[V]": self.magnitude,
+            "i[V]": self.voltage.i,
+            "q[V]": self.voltage.q,
+            "phase[rad]": self.phase,
+        }
+        return serialized_dict
 
     @property
     def average(self):
         """Perform average over i and q"""
-        return AveragedResults.from_components(np.mean(self.i), np.mean(self.q))
+        average_i, average_q = np.array([]), np.array([])
+        std_i, std_q = np.array([]), np.array([])
+        for is_, qs_ in zip(self.voltage.i, self.voltage.q):
+            average_i, average_q = np.append(average_i, np.mean(is_)), np.append(average_q, np.mean(qs_))
+            std_i, std_q = np.append(std_i, np.std(is_)), np.append(std_q, np.std(qs_))
+        return AveragedIQResults(average_i, average_q, std_i=std_i, std_q=std_q)
+
+
+# FIXME: Here I take the states from IQResult that are typed to be ints but those are not what would you do ?
+class AveragedIQResults(IQResults):
+    """
+    Data structure to deal with the output of
+    :func:`qibolab.platforms.abstract.AbstractPlatform.execute_pulse_sequence`
+    :func:`qibolab.platforms.abstract.AbstractPlatform.sweep`
+
+    Associated with AcquisitionType.INTEGRATION and AveragingMode.CYCLIC
+    or the averages of ``IQResults``
+    """
+
+    def __init__(self, i: np.ndarray, q: np.ndarray, shots=None, std_i=None, std_q=None):
+        IQResults.__init__(self, i, q, shots)
+        self.std: Optional[npt.NDArray[np.float64]] = np.recarray(i.shape, dtype=iq_typing)
+        self.std["i"] = std_i
+        self.std["q"] = std_q
+
+
+class RawWaveformResults(IQResults):
+    """
+    Data structure to deal with the output of
+    :func:`qibolab.platforms.abstract.AbstractPlatform.execute_pulse_sequence`
+    :func:`qibolab.platforms.abstract.AbstractPlatform.sweep`
+
+    Associated with AcquisitionType.RAW and AveragingMode.SINGLESHOT
+    may also be used to store the integration weights ?
+    """
+
+
+class AveragedRawWaveformResults(AveragedIQResults):
+    """
+    Data structure to deal with the output of
+    :func:`qibolab.platforms.abstract.AbstractPlatform.execute_pulse_sequence`
+    :func:`qibolab.platforms.abstract.AbstractPlatform.sweep`
+
+    Associated with AcquisitionType.RAW and AveragingMode.CYCLIC
+    or the averages of ``RawWaveformResults``
+    """
+
+
+# FIXME: If probabilities are out of range the error is displeyed weirdly
+@dataclass
+class StateResults:
```

Comment:
Also here we could use
```suggestion
class DiscriminatedResults:
```
but `StateResults` is also fine I think.

Reply:
I didn't really like the name. Apart from sounding bad, it may seems as if you filtered some of the data out. But, yeah I would go with whatever most people prefer.